client/src/hooks/datafiles/mutations/useExtract.ts

```diff
@@ -0,0 +1,126 @@
+import { useMutation } from '@tanstack/react-query';
+import { useSelector, useDispatch, shallowEqual } from 'react-redux';
+import { getExtractParams } from 'utils/getExtractParams';
+import { apiClient } from 'utils/apiClient';
+import { fetchUtil } from 'utils/fetchUtil';
+import { TTapisFile } from 'utils/types';
+import { TJobBody, TJobPostResponse } from './useSubmitJob';
+
+const getAppUtil = async function fetchAppDefinitionUtil(
+  appId: string,
+  appVersion: string
+) {
+  const params = { appId, appVersion };
+  const result = await fetchUtil({
+    url: '/api/workspace/apps',
+    params,
+  });
+  return result.response;
+};
+
+async function submitJobUtil(body: TJobBody) {
+  const res = await apiClient.post<TJobPostResponse>(
+    `/api/workspace/jobs`,
+    body
+  );
+  return res.data.response;
+}
+
+function useExtract() {
+  const dispatch = useDispatch();
+  const status = useSelector(
+    (state: any) => state.files.operationStatus.extract,
+    shallowEqual
+  );
+
+  const setStatus = (newStatus: any) => {
+    dispatch({
+      type: 'DATA_FILES_SET_OPERATION_STATUS',
+      payload: { status: newStatus, operation: 'extract' },
+    });
+  };
+
+  const extractApp = useSelector(
+    (state: any) => state.workbench.config.extractApp
+  );
+
+  const defaultAllocation = useSelector(
+    (state: any) =>
+      state.allocations.portal_alloc || state.allocations.active[0].projectName
+  );
+
+  const latestExtract = getAppUtil(extractApp.id, extractApp.version);
```

Comment:
The mock state would need to be updated to match this new shape:
https://github.com/TACC/Core-Portal/blob/8694ee4922d0a281f516513a5056ff0926b3f130/client/src/components/DataFiles/tests/DataFiles.test.jsx#L18